wordpress-mobile/WordPress-Android
# Use updated referrers store

Fixes #10057 

In order to lower the load there are some changes in the FluxC - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1728

I'm also lowering the number of results on the `View all` page. I think 300 is more than enough results and I don't see anyone needing to scroll lower on a mobile device. Unfortunately, the endpoint doesn't offer pagination. 

To test:
- Go to a site with loads of referrers 
- Open stats/year stats
- Scroll to the Referrers block
- Notice it's showing the correct data (compare to calypso)
- Click on "View more"
- Notice the full screen view is showing the correct data

PR submission checklist:
